lib/handler/configurator/proxy.c

```diff
@@ -99,6 +100,13 @@ static void update_ssl_ctx(SSL_CTX **ctx, X509_STORE *cert_store, int verify_mod
     CRYPTO_add(&cert_store->references, 1, CRYPTO_LOCK_X509_STORE);
     if (verify_mode == -1)
         verify_mode = (*ctx)->verify_mode;
+    if (session_cache == NULL && delete_session_cache == 0) {
+        session_cache = h2o_socket_ssl_get_session_cache(*ctx);
+    }
+    if ((*ctx)->references == 1) {
+        /* ctx will be disposed, but prevent the cache from being disposed by setting NULL to reuse it later*/
+        h2o_socket_ssl_set_session_cache(*ctx, NULL);
+    }
```

Comment:
I don't think this is the right change.

It seems to me that with this change, the function might associate one session cache to more than one SSL_CTX, if `(*ctx)->references` is greater than one.

As discussed in my previous comment, the rule we agreed is that the relation between `SSL_CTX` vs a session cache should be either 1:0 or 1:1.

IMO the easiest way to fulfill such rule would be to create and (optionally) associate the session cache at the time when and only when SSL_CTX is created within this function.